src/rdkafka.h

```diff
@@ -4197,7 +4200,8 @@ RD_EXPORT int rd_kafka_assignment_lost(rd_kafka_t *rk);
 RD_EXPORT rd_kafka_resp_err_t
 rd_kafka_commit(rd_kafka_t *rk,
                 const rd_kafka_topic_partition_list_t *offsets,
-                int async);
+                int async,
+                int timeout_ms);
```

Comment:
These C APIs can't be changed since it would break both the API and ABI compatibility guarantees.

What you would need to do is add new APIs, but this is something we typically try to avoid - to keep the API surface area down and since designing new APIs in a future-proof manner is tricky.

Additionally, I'm not convinced a timeout is the best option here. A timed out commit call could still have sent a commit request to the broker.
I think a better option is to use the existing rd_kafka_commit_queue() API with an callback.

Reply:
Thank you, @edenhill 
I agree on keeping the C APIs unchanged. Apart from that, I'm just wondering if `use the existing rd_kafka_commit_queue() API with an callback` would help, -- e.g. if the brokers are down, and a `sync commit` has just been sent, is there a way to make `rd_kafka_commit` (or, `rd_kafka_commit_queue`) not wait there for ever?